app/src/main/java/com/kms/cura/view/activity/DoctorAppointmentDetailActivity.java

```diff
@@ -0,0 +1,304 @@
+        txtApptDate = loadTextView(R.id.txtAppDay, getApptDate(appointmentEntity.getApptDay()));
+        txtApptTime = loadTextView(R.id.txtApptTime, getApptTime(appointmentEntity.getStartTime(), appointmentEntity.getEndTime()));
+        FacilityEntity facilityEntity = appointmentEntity.getFacilityEntity();
+        txtApptFacility = loadTextView(R.id.txtApptFacility, facilityEntity.getName());
+        txtApptFacilityAddress = loadTextView(R.id.txtApptFacilityAddress, DataUtils.showUnicode(facilityEntity.getAddress()));
+        txtApptFacilityPhone = loadTextView(R.id.txtApptFacilityPhone, facilityEntity.getPhone());
+        loadComment();
+        loadButton();
+    }
+
+    private void loadComment() {
+        String patientCmt = appointmentEntity.getPatientCmt();
+        if (patientCmt == null) {
+            lbExtraComment.setVisibility(View.GONE);
+            return;
+        }
+        txtComment = (TextView) findViewById(R.id.txtComment);
+        txtComment.setText(patientCmt);
+    }
+
+    private AppointmentEntity loadData() {
+        position = getIntent().getIntExtra(CalendarView.APPT_POSITION, -1);
+        if (position == -1) {
+            return null;
+        }
+        AppointmentEntity entity = ((DoctorUserEntity) (CurrentUserProfile.getInstance().getEntity())).getAppointmentList().get(position);
+        return entity;
+    }
+
+    private TextView loadTextView(int id, String text) {
+        TextView textView = (TextView) findViewById(id);
+        textView.setText(text);
+        return textView;
+    }
+
+    private void loadButton() {
+        loadTagButton();
+        btnBack = (ImageView) initButton(R.id.btnBack);
+        btnSendMessage = (Button) initButton(R.id.btnSendMessage);
+        btnCancel = (Button) initButton(R.id.btnCancel);
+        int status = appointmentEntity.getStatus();
+        if (status == AppointmentEntity.INCOMPLETED_STT) {
+            btnComplete = (Button) initButton(R.id.btnCompleteAppt);
+            btnComplete.setVisibility(View.VISIBLE);
+            btnCancel.setText(getResources().getString(R.string.CancelAppt));
+        } else if (status == AppointmentEntity.ACCEPTED_STT){
+            btnCancel.setText(getResources().getString(R.string.CancelAppt));
+        }
+        else {
+            btnCancel.setVisibility(View.GONE);
+        }
+    }
+
+    private View initButton(int id) {
+        View button = findViewById(id);
+        button.setOnClickListener(this);
+        return button;
+    }
+
+    private void loadTagButton() {
+        btnTag = (Button) findViewById(R.id.btnTag);
+        String tag = appointmentEntity.getStatusName();
+        btnTag.setText(tag);
+        Integer color = getTagId(appointmentEntity.getStatus());
+        if (color == null){
+            btnTag.setVisibility(View.GONE);
+            return;
+        }
+        btnTag.setBackgroundResource(color);
+    }
+
+    private Integer getTagId(int status) {
```

Comment:
Then why is line 126 like this `Integer color = getTagId(appointmentEntity.getStatus());` ?